reflect_abi_patch.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"maps"
+	"os"
+	"slices"
+	"strings"
+)
+
+func abiNamePatch(path string) (string, error) {
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return "", err
+	}
+
+	find := `return unsafe.String(n.DataChecked(1+i, "non-empty string"), l)`
+	replace := `name := unsafe.String(n.DataChecked(1+i, "non-empty string"), l)
+	return _realName(name)`
+
+	str := strings.Replace(string(data), find, replace, 1)
+
+	realname := `
+//go:linkname _realName
+func _realName(name string) string
```

Comment:
this needs a TODO for us to obfuscate the name `_realName`, as otherwise it will become incredibly easy to spot and use for obfuscated binaries

Reply:
If replaced the code in a way where
we "just" need to enable obfuscation for "internal/abi" for this to be obfuscated.